src/libs/Navigation/Navigation.js

```diff
@@ -125,13 +125,22 @@ function dismissModal(shouldOpenDrawer = false) {
 /**
  * Check whether the passed route is currently Active or not.
  *
+ * Previous implementation used navigationRef.current.getCurrentRoute().path, which was
+ * undefined in the first navigation. Hence, in our current solution, we're rebuilding
+ * the active route using the useLinkBuilder hook that takes routeName and params as the arguments.
+ *
  * @param {String} routePath Path to check
  * @return {Boolean} is active
  */
-function isActive(routePath) {
+function isActiveRoute(routePath) {
+    const buildLink = useLinkBuilder();
```

Comment:
Apart from https://reactjs.org/docs/hooks-rules.html#only-call-hooks-from-react-functions. All good.

Reply:
Okay. But it seems that using `getPathFromState` will have to also be inside a component only. Any suggestions on this?

React Navigation suggests this approach:
https://reactnavigation.org/docs/screen-tracking/#example

The second approach would then be creating a ref in `Navigation`
`export const currentRouteRef = React.createRef();`

and in NavigationRoot.parseAndStoreRoute

`currentRouteRef.current = path;`

Reply:
It is already using navigationRef in the isActiveRoute.